lineapy/graph_reader/node_collection.py

```diff
@@ -93,6 +93,7 @@ class NodeCollection:
     sliced_nodes: Set[LineaID] = field(default_factory=set)
     raw_codeblock: str = field(default="")
     is_empty: bool = field(default=True)
+    use_cache: Union[None, Tuple[str, Optional[int]]] = field(default=None)
```

Comment:
Change this name, as commented here https://github.com/LineaLabs/lineapy/pull/747/files#r951684298 this use_cache is not the same as it is in ArtifactCollection or SessionArtifact and should not be named the same.

Reply:
My preferred name in this case is to change this use_cache to a boolean since the only operation we are doing is to compare it to None, and call it something like `is_input_artifact`